hedyorg/hedy
# 🪲 Allow comments to be indented #3287

Fixes #3287
The PR addresses multiple small issues related to comments and trailing spaces. Automated tests are added for all scenarios.

**How to test**
1. Indented comments and comments at the end of code blocks should work in levels 9-11 without errors:
```
repeat 3 times
    # comment
    print ‘3’ # another comment
```
2. The `print` command should be able to print a # sign in levels 4-18:
```
print 'comments start with the # sign'
```
3. Equality checks should preserve the original whitespacing of its arguments in levels 8-11. The following program should print the value of `i`.
```
i = this   is
print '[' i ']'
if i = ‘this   is’
  print i
```
